x/twap/logic.go

```diff
@@ -123,6 +123,12 @@ func (k Keeper) EndBlock(ctx sdk.Context) {
 			ctx.Logger().Error("Error pruning old twaps at the end block", err)
 		}
 	}
+
+	// If we still have more deprecated historical twaps to prune, then we prune up to the per block limit.
+	// TODO: Can remove this in the v25 upgrade or after.
+	if k.IsDeprecatedHistoricalTWAPsPruning(ctx) {
```

Comment:
I know you've added some tests to upgrade test, but can we add some more test for this here as well?